### PR TITLE
Add calls' polling entities to the pollset_set of client_channel

### PR DESCRIPTION
**Pull request by y-zeng**
- In client_channel, if `pick_subchannel` thinks a subchannel is not available immediately, add the polling entity in `call_data` to the `interested_parties` of `channel_data`, so that the IO of the lb policy and resolver could be done under it.
- Remove `call_data`s' polling entities from lb policies' `interested_parties`

### CODE REVIEW

**Comment by y-zeng** [comment]
@dgquintas Could you please take a look at the change in lb policies?

**Comment by y-zeng** [comment]
Breaks grpclb

**Comment by y-zeng** [comment]
Ping :grin:

**Comment by dgquintas** [comment]
woops, sorry. I'll have a look tomorrow for sure :)

File: src/core/ext/client_config/lb_policy.h [hunk 1]
```diff
@@ -127,9 +125,9 @@ void grpc_lb_policy_init(grpc_lb_policy *policy,
 /** Given initial metadata in \a initial_metadata, find an appropriate
     target for this rpc, and 'return' it by calling \a on_complete after setting
     \a target.
-    Picking can be asynchronous. Any IO should be done under \a pollent. */
+    Picking can be asynchronous. Any IO should be done under the pollset_set
+    interested_parties. */
```
Comment (dgquintas) [inline src/core/ext/client_config/lb_policy.h]: I don't understand this comment. Do you mean "... under the \a interested_parties \a grpc_pollset_set in the \a grpc_lb_policy struct"? Also, please prefix symbol names with `\a` so that doxygen is best able to automatically format and generate links for them.

File: src/core/ext/client_config/client_channel.c [hunk 2]
```diff
@@ -456,10 +456,14 @@ static void retry_waiting_locked(grpc_exec_ctx *exec_ctx, call_data *calld) {
...
 static void subchannel_ready(grpc_exec_ctx *exec_ctx, void *arg,
                              grpc_error *error) {
-  call_data *calld = arg;
+  grpc_call_element *elem = arg;
+  call_data *calld = elem->call_data;
+  channel_data *chand = elem->channel_data;
   gpr_mu_lock(&calld->mu);
   GPR_ASSERT(calld->creation_phase ==
              GRPC_SUBCHANNEL_CALL_HOLDER_PICKING_SUBCHANNEL);
+  grpc_polling_entity_del_from_pollset_set(exec_ctx, calld->pollent,
```
Comment (dgquintas) [inline src/core/ext/client_config/client_channel.c]: out of curiosity: all the removed `grpc_polling_entity_del_from_pollset_set` invocations from the LB policies are now handled by this one?

File: src/core/ext/lb_policy/grpclb/grpclb.c [hunk 3]
```diff
@@ -769,7 +761,7 @@ static lb_client_data *lb_client_data_create(glb_lb_policy *glb_policy) {
 
   /* Note the following LB call progresses every time there's activity in \a
    * glb_policy->base.interested_parties, which is comprised of the polling
-   * entities passed to glb_pick(). */
+   * entities from client_channel. */
```
Comment (dgquintas) [inline src/core/ext/lb_policy/grpclb/grpclb.c]: \a client_channel

File: src/core/ext/client_config/lb_policy.h (hunk 1 above)
Comment (y-zeng) [inline src/core/ext/client_config/lb_policy.h]: Thanks a lot for the review!
I was trying to say "... under a \a grpc_pollset_set called \a interested_parties". But "... under the \a interested_parties \a grpc_pollset_set in the \a grpc_lb_policy struct" seems better .

File: src/core/ext/client_config/client_channel.c (hunk 2 above)
Comment (y-zeng) [inline src/core/ext/client_config/client_channel.c]: Yes. The removed `grpc_polling_entity_del_from_pollset_set` invocations are followed by `grpc_exec_ctx_sched(exec_ctx, pp->on_complete, ...)`, and this one will be invoked in the `on_complete` closure.

File: src/core/ext/lb_policy/grpclb/grpclb.c (hunk 3 above)
Comment (y-zeng) [inline src/core/ext/lb_policy/grpclb/grpclb.c]: Fixed.

**Comment by dgquintas** [comment]
@y-zeng feel free to merge once the tests pass.

**Comment by y-zeng** [comment]
@dgquintas Thanks! I'm still rerunning tests to make sure suspicious failures were not caused by this PR.

**Comment by ctiller** [comment]
Needs a merge.

Please at least eyeball the performance smoke tests before submission.

**Comment by y-zeng** [comment]
The performance smoke test results are as follows (ran locally):

Master:
```
RUNNING SCENARIO: cpp_generic_async_streaming_ping_pong_secure
I1005 16:09:25.628789938   59028 report.cc:75]               QPS: 10231.0
I1005 16:09:25.632208597   59028 report.cc:79]               QPS: 10231.0 (10231.0/server core)
I1005 16:09:25.632248398   59028 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 92.3/121.7/123.5/162.1/441.8 us
I1005 16:09:25.632262156   59028 report.cc:94]               Server system time: 34.17%
I1005 16:09:25.632272798   59028 report.cc:96]               Server user time:   25.38%
I1005 16:09:25.632282851   59028 report.cc:98]               Client system time: 32.49%
I1005 16:09:25.632292760   59028 report.cc:100]              Client user time:   27.46%

RUNNING SCENARIO: cpp_generic_async_streaming_qps_unconstrained_secure
I1005 16:10:06.303957907   59251 report.cc:75]               QPS: 781932.2
I1005 16:10:06.306036214   59251 report.cc:79]               QPS: 781932.2 (65161.0/server core)
I1005 16:10:06.306120866   59251 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 7208.5/11808.1/14508.6/21484.3/38461.4 us
I1005 16:10:06.306135890   59251 report.cc:94]               Server system time: 35.22%
I1005 16:10:06.306147166   59251 report.cc:96]               Server user time:   510.65%
I1005 16:10:06.306157520   59251 report.cc:98]               Client system time: 34.52%
I1005 16:10:06.306167767   59251 report.cc:100]              Client user time:   485.39%

RUNNING SCENARIO: cpp_generic_async_streaming_qps_one_server_core_secure
I1005 16:10:45.994091652   59410 report.cc:75]               QPS: 223053.8
I1005 16:10:45.995060558   59410 report.cc:79]               QPS: 223053.8 (223053.8/server core)
I1005 16:10:45.995075617   59410 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 28031.7/30423.8/31602.7/35360.7/98617.3 us
I1005 16:10:45.995080052   59410 report.cc:94]               Server system time: 6.41%
I1005 16:10:45.995083823   59410 report.cc:96]               Server user time:   93.29%
I1005 16:10:45.995087326   59410 report.cc:98]               Client system time: 15.01%
I1005 16:10:45.995090847   59410 report.cc:100]              Client user time:   159.96%

RUNNING SCENARIO: cpp_protobuf_sync_unary_ping_pong_secure
I1005 16:11:22.557751351   59549 report.cc:75]               QPS: 7252.9
I1005 16:11:22.559042139   59549 report.cc:79]               QPS: 7252.9 (7252.9/server core)
I1005 16:11:22.559061809   59549 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 132.2/153.1/163.0/215.7/764.2 us
I1005 16:11:22.559068971   59549 report.cc:94]               Server system time: 30.12%
I1005 16:11:22.559075030   59549 report.cc:96]               Server user time:   31.44%
I1005 16:11:22.559080726   59549 report.cc:98]               Client system time: 25.62%
I1005 16:11:22.559086279   59549 report.cc:100]              Client user time:   30.20%

RUNNING SCENARIO: cpp_protobuf_sync_unary_qps_unconstrained_secure
I1005 16:12:02.975495489   59617 report.cc:75]               QPS: 20739.4
I1005 16:12:02.976423956   59617 report.cc:79]               QPS: 20739.4 (1728.3/server core)
I1005 16:12:02.976438194   59617 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 46804.2/58287.9/64936.5/106739.7/151050.0 us
I1005 16:12:02.976442905   59617 report.cc:94]               Server system time: 161.49%
I1005 16:12:02.976446750   59617 report.cc:96]               Server user time:   121.99%
I1005 16:12:02.976450426   59617 report.cc:98]               Client system time: 377.94%
I1005 16:12:02.976454082   59617 report.cc:100]              Client user time:   132.12%

RUNNING SCENARIO: cpp_protobuf_async_unary_ping_pong_secure
I1005 16:12:42.660151588   86358 report.cc:75]               QPS: 6662.3
I1005 16:12:42.662182298   86358 report.cc:79]               QPS: 6662.3 (6662.3/server core)
I1005 16:12:42.662221322   86358 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 157.1/175.6/179.7/222.8/413.2 us
I1005 16:12:42.662242066   86358 report.cc:94]               Server system time: 24.63%
I1005 16:12:42.662258938   86358 report.cc:96]               Server user time:   32.23%
I1005 16:12:42.662275409   86358 report.cc:98]               Client system time: 25.55%
I1005 16:12:42.662288229   86358 report.cc:100]              Client user time:   33.82%

RUNNING SCENARIO: cpp_protobuf_async_unary_qps_unconstrained_secure
I1005 16:13:23.657500329   86424 report.cc:75]               QPS: 321724.6
I1005 16:13:23.659672409   86424 report.cc:79]               QPS: 321724.6 (26810.4/server core)
I1005 16:13:23.659702441   86424 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 18056.0/27426.2/32810.2/44472.9/64670.4 us
I1005 16:13:23.659715257   86424 report.cc:94]               Server system time: 15.62%
I1005 16:13:23.659726085   86424 report.cc:96]               Server user time:   464.55%
I1005 16:13:23.659736276   86424 report.cc:98]               Client system time: 18.24%
I1005 16:13:23.659746442   86424 report.cc:100]              Client user time:   593.00%

RUNNING SCENARIO: cpp_protobuf_sync_streaming_ping_pong_secure
I1005 16:14:02.649005575   86709 report.cc:75]               QPS: 8878.7
I1005 16:14:02.651248782   86709 report.cc:79]               QPS: 8878.7 (8878.7/server core)
I1005 16:14:02.651281902   86709 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 114.5/125.0/128.0/145.5/287.6 us
I1005 16:14:02.651294939   86709 report.cc:94]               Server system time: 35.72%
I1005 16:14:02.651305657   86709 report.cc:96]               Server user time:   29.74%
I1005 16:14:02.651315720   86709 report.cc:98]               Client system time: 31.47%
I1005 16:14:02.651325780   86709 report.cc:100]              Client user time:   28.02%

RUNNING SCENARIO: cpp_protobuf_sync_streaming_qps_unconstrained_secure
I1005 16:14:43.378070479   86797 report.cc:75]               QPS: 36860.8
I1005 16:14:43.380301835   86797 report.cc:79]               QPS: 36860.8 (3071.7/server core)
I1005 16:14:43.380335550   86797 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 28103.4/36656.7/39698.5/46831.1/57561.9 us
I1005 16:14:43.380348586   86797 report.cc:94]               Server system time: 177.50%
I1005 16:14:43.380359388   86797 report.cc:96]               Server user time:   81.95%
I1005 16:14:43.380369669   86797 report.cc:98]               Client system time: 412.63%
I1005 16:14:43.380379558   86797 report.cc:100]              Client user time:   113.77%

RUNNING SCENARIO: cpp_protobuf_async_streaming_ping_pong_secure
I1005 16:15:25.742805456   88968 report.cc:75]               QPS: 9816.1
I1005 16:15:25.745037230   88968 report.cc:79]               QPS: 9816.1 (9816.1/server core)
I1005 16:15:25.745069708   88968 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 96.0/127.8/130.3/163.9/308.9 us
I1005 16:15:25.745083019   88968 report.cc:94]               Server system time: 32.51%
I1005 16:15:25.745093904   88968 report.cc:96]               Server user time:   26.89%
I1005 16:15:25.745104051   88968 report.cc:98]               Client system time: 33.66%
I1005 16:15:25.745114094   88968 report.cc:100]              Client user time:   27.05%

RUNNING SCENARIO: cpp_protobuf_async_streaming_qps_unconstrained_secure
I1005 16:16:06.671629530   89024 report.cc:75]               QPS: 797879.8
I1005 16:16:06.673856102   89024 report.cc:79]               QPS: 797879.8 (66490.0/server core)
I1005 16:16:06.673887136   89024 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 7387.9/10300.3/12541.2/18592.9/31059.3 us
I1005 16:16:06.673900327   89024 report.cc:94]               Server system time: 38.97%
I1005 16:16:06.673910938   89024 report.cc:96]               Server user time:   515.71%
I1005 16:16:06.673921118   89024 report.cc:98]               Client system time: 35.73%
I1005 16:16:06.673931305   89024 report.cc:100]              Client user time:   534.90%

RUNNING SCENARIO: cpp_generic_async_streaming_qps_one_server_core_insecure
I1005 16:16:45.933643555   89179 report.cc:75]               QPS: 246356.3
I1005 16:16:45.935835300   89179 report.cc:79]               QPS: 246356.3 (246356.3/server core)
I1005 16:16:45.935867832   89179 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 25552.7/27218.5/28094.3/32108.6/84397.6 us
I1005 16:16:45.935879963   89179 report.cc:94]               Server system time: 6.46%
I1005 16:16:45.935890574   89179 report.cc:96]               Server user time:   93.28%
I1005 16:16:45.935900654   89179 report.cc:98]               Client system time: 16.69%
I1005 16:16:45.935910931   89179 report.cc:100]              Client user time:   156.31%

RUNNING SCENARIO: cpp_protobuf_sync_unary_ping_pong_insecure
I1005 16:17:22.775819776   89309 report.cc:75]               QPS: 8563.3
I1005 16:17:22.778022493   89309 report.cc:79]               QPS: 8563.3 (8563.3/server core)
I1005 16:17:22.778053732   89309 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 115.4/127.6/131.9/157.3/324.7 us
I1005 16:17:22.778066294   89309 report.cc:94]               Server system time: 35.17%
I1005 16:17:22.778076554   89309 report.cc:96]               Server user time:   31.29%
I1005 16:17:22.778086377   89309 report.cc:98]               Client system time: 29.18%
I1005 16:17:22.778096123   89309 report.cc:100]              Client user time:   26.97%

RUNNING SCENARIO: cpp_protobuf_async_unary_ping_pong_insecure
I1005 16:18:02.847378383   89519 report.cc:75]               QPS: 7923.1
I1005 16:18:02.849590482   89519 report.cc:79]               QPS: 7923.1 (7923.1/server core)
I1005 16:18:02.849628984   89519 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 130.5/141.6/144.4/177.0/318.4 us
I1005 16:18:02.849649454   89519 report.cc:94]               Server system time: 29.89%
I1005 16:18:02.849661078   89519 report.cc:96]               Server user time:   30.42%
I1005 16:18:02.849671151   89519 report.cc:98]               Client system time: 30.48%
I1005 16:18:02.849681161   89519 report.cc:100]              Client user time:   29.04%

RUNNING SCENARIO: cpp_protobuf_sync_streaming_ping_pong_insecure
I1005 16:18:42.807628073   89574 report.cc:75]               QPS: 11493.4
I1005 16:18:42.808499842   89574 report.cc:79]               QPS: 11493.4 (11493.4/server core)
I1005 16:18:42.808512039   89574 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 87.5/93.1/97.8/136.3/243.2 us
I1005 16:18:42.808516605   89574 report.cc:94]               Server system time: 42.82%
I1005 16:18:42.808520260   89574 report.cc:96]               Server user time:   26.04%
I1005 16:18:42.808523679   89574 report.cc:98]               Client system time: 37.10%
I1005 16:18:42.808527120   89574 report.cc:100]              Client user time:   23.72%

RUNNING SCENARIO: cpp_protobuf_async_streaming_ping_pong_insecure
I1005 16:19:25.866114788   89628 report.cc:75]               QPS: 10759.3
I1005 16:19:25.868331309   89628 report.cc:79]               QPS: 10759.3 (10759.3/server core)
I1005 16:19:25.868364022   89628 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 95.0/99.1/101.9/114.5/186.0 us
I1005 16:19:25.868376874   89628 report.cc:94]               Server system time: 38.01%
I1005 16:19:25.868387499   89628 report.cc:96]               Server user time:   23.87%
I1005 16:19:25.868397419   89628 report.cc:98]               Client system time: 37.85%
I1005 16:19:25.868407264   89628 report.cc:100]              Client user time:   24.24%
```

This PR:
```
RUNNING SCENARIO: cpp_generic_async_streaming_ping_pong_secure
I1005 15:47:53.311832195   26969 report.cc:75]               QPS: 10124.4
I1005 15:47:53.315490518   26969 report.cc:79]               QPS: 10124.4 (10124.4/server core)
I1005 15:47:53.315535024   26969 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 95.4/121.3/123.0/153.1/348.6 us
I1005 15:47:53.315553246   26969 report.cc:94]               Server system time: 32.63%
I1005 15:47:53.315564148   26969 report.cc:96]               Server user time:   26.62%
I1005 15:47:53.315574381   26969 report.cc:98]               Client system time: 32.87%
I1005 15:47:53.315584625   26969 report.cc:100]              Client user time:   28.50%

RUNNING SCENARIO: cpp_generic_async_streaming_qps_unconstrained_secure
I1005 15:48:34.339709959   27048 report.cc:75]               QPS: 737709.2
I1005 15:48:34.341791571   27048 report.cc:79]               QPS: 737709.2 (61475.8/server core)
I1005 15:48:34.341822287   27048 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 7594.4/13631.6/16394.2/22765.2/36090.6 us
I1005 15:48:34.341833295   27048 report.cc:94]               Server system time: 33.41%
I1005 15:48:34.341842689   27048 report.cc:96]               Server user time:   485.80%
I1005 15:48:34.341851417   27048 report.cc:98]               Client system time: 33.03%
I1005 15:48:34.341860123   27048 report.cc:100]              Client user time:   467.32%

RUNNING SCENARIO: cpp_generic_async_streaming_qps_one_server_core_secure
I1005 15:49:13.655735486   27212 report.cc:75]               QPS: 228474.2
I1005 15:49:13.658089508   27212 report.cc:79]               QPS: 228474.2 (228474.2/server core)
I1005 15:49:13.658123773   27212 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 27336.8/29548.2/31097.4/38730.9/95923.3 us
I1005 15:49:13.658136743   27212 report.cc:94]               Server system time: 5.76%
I1005 15:49:13.658148012   27212 report.cc:96]               Server user time:   93.90%
I1005 15:49:13.658158634   27212 report.cc:98]               Client system time: 16.06%
I1005 15:49:13.658169515   27212 report.cc:100]              Client user time:   167.25%

RUNNING SCENARIO: cpp_protobuf_sync_unary_ping_pong_secure
I1005 15:49:50.246750890   27439 report.cc:75]               QPS: 6618.0
I1005 15:49:50.248890625   27439 report.cc:79]               QPS: 6618.0 (6618.0/server core)
I1005 15:49:50.248921423   27439 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 146.2/172.7/178.1/230.6/588.9 us
I1005 15:49:50.248934302   27439 report.cc:94]               Server system time: 29.91%
I1005 15:49:50.248945378   27439 report.cc:96]               Server user time:   32.52%
I1005 15:49:50.248955628   27439 report.cc:98]               Client system time: 25.11%
I1005 15:49:50.248965828   27439 report.cc:100]              Client user time:   30.40%

RUNNING SCENARIO: cpp_protobuf_sync_unary_qps_unconstrained_secure
I1005 15:50:30.641452837   27596 report.cc:75]               QPS: 20489.5
I1005 15:50:30.643000437   27596 report.cc:79]               QPS: 20489.5 (1707.5/server core)
I1005 15:50:30.643018242   27596 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 46751.1/58793.2/67325.8/124989.2/159068.5 us
I1005 15:50:30.643024633   27596 report.cc:94]               Server system time: 156.13%
I1005 15:50:30.643029763   27596 report.cc:96]               Server user time:   120.83%
I1005 15:50:30.643034550   27596 report.cc:98]               Client system time: 357.54%
I1005 15:50:30.643039282   27596 report.cc:100]              Client user time:   129.39%

RUNNING SCENARIO: cpp_protobuf_async_unary_ping_pong_secure
I1005 15:51:10.330272611   50819 report.cc:75]               QPS: 7357.5
I1005 15:51:10.332675308   50819 report.cc:79]               QPS: 7357.5 (7357.5/server core)
I1005 15:51:10.332720242   50819 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 137.3/173.8/176.0/198.7/337.5 us
I1005 15:51:10.332738942   50819 report.cc:94]               Server system time: 23.77%
I1005 15:51:10.332754754   50819 report.cc:96]               Server user time:   32.60%
I1005 15:51:10.332771179   50819 report.cc:98]               Client system time: 25.19%
I1005 15:51:10.332787423   50819 report.cc:100]              Client user time:   34.55%

RUNNING SCENARIO: cpp_protobuf_async_unary_qps_unconstrained_secure
I1005 15:51:51.258327064   51079 report.cc:75]               QPS: 331636.3
I1005 15:51:51.260543443   51079 report.cc:79]               QPS: 331636.3 (27636.4/server core)
I1005 15:51:51.260585449   51079 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 17741.6/24981.7/30285.3/41636.6/57183.4 us
I1005 15:51:51.260605940   51079 report.cc:94]               Server system time: 16.92%
I1005 15:51:51.260623711   51079 report.cc:96]               Server user time:   480.42%
I1005 15:51:51.260640296   51079 report.cc:98]               Client system time: 18.64%
I1005 15:51:51.260656500   51079 report.cc:100]              Client user time:   607.99%

RUNNING SCENARIO: cpp_protobuf_sync_streaming_ping_pong_secure
I1005 15:52:30.339447314   51205 report.cc:75]               QPS: 11519.3
I1005 15:52:30.341610376   51205 report.cc:79]               QPS: 11519.3 (11519.3/server core)
I1005 15:52:30.341641207   51205 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 84.0/99.2/105.2/132.7/234.3 us
I1005 15:52:30.341655226   51205 report.cc:94]               Server system time: 36.84%
I1005 15:52:30.341666987   51205 report.cc:96]               Server user time:   25.99%
I1005 15:52:30.341678486   51205 report.cc:98]               Client system time: 33.27%
I1005 15:52:30.341689806   51205 report.cc:100]              Client user time:   27.34%

RUNNING SCENARIO: cpp_protobuf_sync_streaming_qps_unconstrained_secure
I1005 15:53:10.972096141   51283 report.cc:75]               QPS: 37450.0
I1005 15:53:10.973788542   51283 report.cc:79]               QPS: 37450.0 (3120.8/server core)
I1005 15:53:10.973811021   51283 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 27712.9/36102.3/38966.9/45452.2/56037.9 us
I1005 15:53:10.973819014   51283 report.cc:94]               Server system time: 180.06%
I1005 15:53:10.973825459   51283 report.cc:96]               Server user time:   83.81%
I1005 15:53:10.973831809   51283 report.cc:98]               Client system time: 415.56%
I1005 15:53:10.973838930   51283 report.cc:100]              Client user time:   115.19%

RUNNING SCENARIO: cpp_protobuf_async_streaming_ping_pong_secure
I1005 15:53:53.436493419   53576 report.cc:75]               QPS: 10592.2
I1005 15:53:53.438736081   53576 report.cc:79]               QPS: 10592.2 (10592.2/server core)
I1005 15:53:53.438767771   53576 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 92.0/127.5/129.3/145.0/314.0 us
I1005 15:53:53.438780804   53576 report.cc:94]               Server system time: 32.49%
I1005 15:53:53.438791369   53576 report.cc:96]               Server user time:   26.61%
I1005 15:53:53.438801151   53576 report.cc:98]               Client system time: 32.56%
I1005 15:53:53.438810840   53576 report.cc:100]              Client user time:   27.67%

RUNNING SCENARIO: cpp_protobuf_async_streaming_qps_unconstrained_secure
I1005 15:54:34.353811119   53720 report.cc:75]               QPS: 745256.2
I1005 15:54:34.355999212   53720 report.cc:79]               QPS: 745256.2 (62104.7/server core)
I1005 15:54:34.356030334   53720 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 7642.9/12051.9/15007.7/21761.2/35818.0 us
I1005 15:54:34.356043036   53720 report.cc:94]               Server system time: 36.04%
I1005 15:54:34.356053697   53720 report.cc:96]               Server user time:   472.17%
I1005 15:54:34.356063653   53720 report.cc:98]               Client system time: 34.79%
I1005 15:54:34.356073563   53720 report.cc:100]              Client user time:   536.78%

RUNNING SCENARIO: cpp_generic_async_streaming_qps_one_server_core_insecure
I1005 15:55:13.640439208   53888 report.cc:75]               QPS: 247312.5
I1005 15:55:13.641692310   53888 report.cc:79]               QPS: 247312.5 (247312.5/server core)
I1005 15:55:13.641708263   53888 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 25396.6/26919.1/28232.9/34930.7/85140.0 us
I1005 15:55:13.641712778   53888 report.cc:94]               Server system time: 6.67%
I1005 15:55:13.641716534   53888 report.cc:96]               Server user time:   93.04%
I1005 15:55:13.641720144   53888 report.cc:98]               Client system time: 17.60%
I1005 15:55:13.641747914   53888 report.cc:100]              Client user time:   155.81%

RUNNING SCENARIO: cpp_protobuf_sync_unary_ping_pong_insecure
I1005 15:55:50.456989808   54118 report.cc:75]               QPS: 8805.2
I1005 15:55:50.458440347   54118 report.cc:79]               QPS: 8805.2 (8805.2/server core)
I1005 15:55:50.458462967   54118 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 112.0/127.6/129.5/148.5/252.1 us
I1005 15:55:50.458470924   54118 report.cc:94]               Server system time: 35.45%
I1005 15:55:50.458477529   54118 report.cc:96]               Server user time:   30.73%
I1005 15:55:50.458483633   54118 report.cc:98]               Client system time: 30.09%
I1005 15:55:50.458489683   54118 report.cc:100]              Client user time:   25.74%

RUNNING SCENARIO: cpp_protobuf_async_unary_ping_pong_insecure
I1005 15:56:30.527182490   54198 report.cc:75]               QPS: 7525.2
I1005 15:56:30.528142363   54198 report.cc:79]               QPS: 7525.2 (7525.2/server core)
I1005 15:56:30.528157893   54198 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 137.5/142.6/145.7/163.0/314.5 us
I1005 15:56:30.528162587   54198 report.cc:94]               Server system time: 29.85%
I1005 15:56:30.528166137   54198 report.cc:96]               Server user time:   30.25%
I1005 15:56:30.528169537   54198 report.cc:98]               Client system time: 29.14%
I1005 15:56:30.528172885   54198 report.cc:100]              Client user time:   30.70%

RUNNING SCENARIO: cpp_protobuf_sync_streaming_ping_pong_insecure
I1005 15:57:10.496126879   54239 report.cc:75]               QPS: 12334.8
I1005 15:57:10.497564315   54239 report.cc:79]               QPS: 12334.8 (12334.8/server core)
I1005 15:57:10.497595547   54239 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 84.7/92.6/95.1/111.4/280.1 us
I1005 15:57:10.497609002   54239 report.cc:94]               Server system time: 43.46%
I1005 15:57:10.497619683   54239 report.cc:96]               Server user time:   24.15%
I1005 15:57:10.497629569   54239 report.cc:98]               Client system time: 37.81%
I1005 15:57:10.497639422   54239 report.cc:100]              Client user time:   22.53%

RUNNING SCENARIO: cpp_protobuf_async_streaming_ping_pong_insecure
I1005 15:57:56.560498219   54492 report.cc:75]               QPS: 11752.2
I1005 15:57:56.562317068   54492 report.cc:79]               QPS: 11752.2 (11752.2/server core)
I1005 15:57:56.562335319   54492 report.cc:84]               Latencies (50/90/95/99/99.9%-ile): 87.6/98.7/100.4/114.6/218.5 us
I1005 15:57:56.562341096   54492 report.cc:94]               Server system time: 37.51%
I1005 15:57:56.562345765   54492 report.cc:96]               Server user time:   23.95%
I1005 15:57:56.562350176   54492 report.cc:98]               Client system time: 37.23%
I1005 15:57:56.562354622   54492 report.cc:100]              Client user time:   24.80%
```

**Comment by dgquintas** [comment]
@y-zeng so that only one person needs to compare them, can you summarize if performance seems unaffected by your change?

**Comment by y-zeng** [comment]
QPS in `cpp_protobuf_sync_unary_ping_pong_secure`, `cpp_protobuf_async_streaming_qps_unconstrained_secure`, and
`cpp_protobuf_async_unary_ping_pong_insecure` are slightly smaller than those in master. But the proportion of client system time did not change significantly (this PR should only affect the client system time).
And following runs showed that the QPS in these tests fluctuated around the numbers in master.

**Comment by y-zeng** [comment]
@dgquintas Yes, the performance seems unaffected by this PR.

**Comment by y-zeng** [comment]
Test failures seem irrelevant.
